### PR TITLE
apps/testcase/systemio: add new uart scenarios TC

### PR DESCRIPTION
**Pull request by samegautam**
File Path : apps/examples/testcase/ta_tc/systemio/itc/itc_uart.c 
Description : add new scenarios testcase for UART

Scenarios testcase name :
itc_iotbus_uart_init_stop_multi_handle_p
itc_iotbus_uart_set_baudrate_n
itc_iotbus_uart_set_mode_all_p
itc_iotbus_uart_write_read_flush_multi_handle_p
itc_iotbus_uart_write_read_flush_p
itc_iotbus_uart_write_read_flush_read_p
itc_iotbus_uart_set_flow_write_read_p
itc_iotbus_uart_set_flow_flush_write_read_p
itc_iotbus_uart_set_mode_write_read_p
itc_iotbus_uart_set_mode_flush_write_read_p
itc_iotbus_uart_set_flow_mode_flush_write_read_p

### CODE REVIEW

**Comment by sunghan-chang** [comment]
@samegautam Please fill details on commit message.  What scenarios do you add?

**Comment by seinfra** [comment]
Code Rule Check Result:
apps/examples/testcase/ta_tc/systemio/itc/itc_uart.c:81: ERROR: [SPC_M_OPR] space required after that ';' (ctx:VxV)
apps/examples/testcase/ta_tc/systemio/itc/itc_uart.c:84: ERROR: [SPC_M_OPR] space required after that ';' (ctx:VxV)
apps/examples/testcase/ta_tc/systemio/itc/itc_uart.c:265: ERROR: [SPC_M_KWD] space required before the open parenthesis '('
apps/examples/testcase/ta_tc/systemio/itc/itc_uart.c:392: ERROR: [SPC_M_OPR] space required after that ';' (ctx:VxV)
apps/examples/testcase/ta_tc/systemio/itc/itc_uart.c:395: ERROR: [SPC_M_OPR] space required after that ';' (ctx:VxV)
apps/examples/testcase/ta_tc/systemio/itc/itc_uart.c:400: ERROR: [SPC_M_OPR] space required after that ';' (ctx:VxV)
apps/examples/testcase/ta_tc/systemio/itc/itc_uart.c:401: ERROR: [SPC_M_OPR] space required after that ';' (ctx:VxV)
apps/examples/testcase/ta_tc/systemio/itc/itc_uart.c:404: ERROR: [SPC_M_OPR] space required after that ';' (ctx:VxV)
apps/examples/testcase/ta_tc/systemio/itc/itc_uart.c:405: ERROR: [SPC_M_OPR] space required after that ';' (ctx:VxV)
apps/examples/testcase/ta_tc/systemio/itc/itc_uart.c:408: ERROR: [SPC_M_OPR] space required after that ';' (ctx:VxV)
apps/examples/testcase/ta_tc/systemio/itc/itc_uart.c:411: ERROR: [SPC_M_OPR] space required after that ';' (ctx:VxV)

**Comment by seinfra** [comment]
Code Rule Check Result:
apps/examples/testcase/ta_tc/systemio/itc/itc_uart.c:81: ERROR: [SPC_M_OPR] space required after that ';' (ctx:VxV)
apps/examples/testcase/ta_tc/systemio/itc/itc_uart.c:84: ERROR: [SPC_M_OPR] space required after that ';' (ctx:VxV)
apps/examples/testcase/ta_tc/systemio/itc/itc_uart.c:265: ERROR: [SPC_M_KWD] space required before the open parenthesis '('
apps/examples/testcase/ta_tc/systemio/itc/itc_uart.c:392: ERROR: [SPC_M_OPR] space required after that ';' (ctx:VxV)
apps/examples/testcase/ta_tc/systemio/itc/itc_uart.c:395: ERROR: [SPC_M_OPR] space required after that ';' (ctx:VxV)
apps/examples/testcase/ta_tc/systemio/itc/itc_uart.c:400: ERROR: [SPC_M_OPR] space required after that ';' (ctx:VxV)
apps/examples/testcase/ta_tc/systemio/itc/itc_uart.c:401: ERROR: [SPC_M_OPR] space required after that ';' (ctx:VxV)
apps/examples/testcase/ta_tc/systemio/itc/itc_uart.c:404: ERROR: [SPC_M_OPR] space required after that ';' (ctx:VxV)
apps/examples/testcase/ta_tc/systemio/itc/itc_uart.c:405: ERROR: [SPC_M_OPR] space required after that ';' (ctx:VxV)
apps/examples/testcase/ta_tc/systemio/itc/itc_uart.c:408: ERROR: [SPC_M_OPR] space required after that ';' (ctx:VxV)
apps/examples/testcase/ta_tc/systemio/itc/itc_uart.c:411: ERROR: [SPC_M_OPR] space required after that ';' (ctx:VxV)

File: apps/examples/testcase/ta_tc/systemio/itc/itc_uart.c [hunk 1]
```diff
@@ -62,6 +60,39 @@ void itc_iotbus_uart_init_stop_p(void)
 }
 
 /**
+* @testcase         itc_iotbus_uart_init_stop_p1
```
[the remark this answers is not in the document]
Comment (samegautam) [reply apps/examples/testcase/ta_tc/systemio/itc/itc_uart.c]: @pradeepns85  : Please review the updated comments of the testcase.

[the remark this answers is not in the document]
Comment (samegautam) [reply apps/examples/testcase/ta_tc/systemio/itc/itc_uart.c]: @pradeepns85 : Please review updated comment of the testcase.

**Comment by seinfra** [comment]
Code Rule Check Result:
apps/examples/testcase/ta_tc/systemio/itc/itc_uart.c:81: ERROR: [SPC_M_OPR] space required after that ';' (ctx:VxV)
apps/examples/testcase/ta_tc/systemio/itc/itc_uart.c:84: ERROR: [SPC_M_OPR] space required after that ';' (ctx:VxV)
apps/examples/testcase/ta_tc/systemio/itc/itc_uart.c:265: ERROR: [SPC_M_KWD] space required before the open parenthesis '('
apps/examples/testcase/ta_tc/systemio/itc/itc_uart.c:392: ERROR: [SPC_M_OPR] space required after that ';' (ctx:VxV)
apps/examples/testcase/ta_tc/systemio/itc/itc_uart.c:395: ERROR: [SPC_M_OPR] space required after that ';' (ctx:VxV)
apps/examples/testcase/ta_tc/systemio/itc/itc_uart.c:400: ERROR: [SPC_M_OPR] space required after that ';' (ctx:VxV)
apps/examples/testcase/ta_tc/systemio/itc/itc_uart.c:401: ERROR: [SPC_M_OPR] space required after that ';' (ctx:VxV)
apps/examples/testcase/ta_tc/systemio/itc/itc_uart.c:404: ERROR: [SPC_M_OPR] space required after that ';' (ctx:VxV)
apps/examples/testcase/ta_tc/systemio/itc/itc_uart.c:405: ERROR: [SPC_M_OPR] space required after that ';' (ctx:VxV)
apps/examples/testcase/ta_tc/systemio/itc/itc_uart.c:408: ERROR: [SPC_M_OPR] space required after that ';' (ctx:VxV)
apps/examples/testcase/ta_tc/systemio/itc/itc_uart.c:411: ERROR: [SPC_M_OPR] space required after that ';' (ctx:VxV)

**Comment by seinfra** [comment]
Code Rule Check Result:
apps/examples/testcase/ta_tc/systemio/itc/itc_uart.c:395: ERROR: [SPC_M_OPR] space required after that ';' (ctx:VxV)

**Comment by seinfra** [comment]
Code Rule Check OK.

File: apps/examples/testcase/ta_tc/systemio/itc/itc_uart.c [hunk 2]
```diff
@@ -62,6 +60,39 @@ void itc_iotbus_uart_init_stop_p(void)
...
+	int ret = IOTBUS_ERROR_NONE;
+
+	iotbus_uart_context_h h_uart1 = iotbus_uart_init(DEVPATH);
+	TC_ASSERT_NEQ("iotbus_uart_init1", h_uart1, NULL);
+
+	iotbus_uart_context_h h_uart2 = iotbus_uart_init(DEVPATH);
+	TC_ASSERT_NEQ_CLEANUP("iotbus_uart_init2", h_uart2, NULL, iotbus_uart_stop(h_uart1));
+
+	iotbus_uart_context_h h_uart3 = iotbus_uart_init(DEVPATH);
+	TC_ASSERT_NEQ_CLEANUP("iotbus_uart_init3", h_uart3, NULL, iotbus_uart_stop(h_uart2); iotbus_uart_stop(h_uart3));
```
Comment (sunghan-chang) [inline apps/examples/testcase/ta_tc/systemio/itc/itc_uart.c]: We should stop 1 and 2 in failure. right?

**Comment by seinfra** [comment]
Code Rule Check OK.

File: apps/examples/testcase/ta_tc/systemio/itc/itc_uart.c (hunk 2 above)
Comment (samegautam) [reply apps/examples/testcase/ta_tc/systemio/itc/itc_uart.c]: @sunghan-chang  
Yes , fixed.

**Comment by samegautam** [comment]
Please review @jeongchanKim @sunghan-chang @junmin-kim @Taejun-Kwon

File: apps/examples/testcase/ta_tc/systemio/itc/itc_uart.c [hunk 3]
```diff
@@ -61,6 +59,39 @@ void itc_iotbus_uart_init_stop_p(void)
...
 
+/**
+* @testcase         itc_iotbus_uart_init_stop_multi_handle_p
+* @brief            initializes and closes uart_context
+* @scenario         initializes and closes uart_context
+* @apicovered       iotbus_uart_init, iotbus_uart_stop
+* @precondition     none
+* @postcondition    none
+*/
+void itc_iotbus_uart_init_stop_multi_handle_p(void)
```
Comment (sunghan-chang) [inline apps/examples/testcase/ta_tc/systemio/itc/itc_uart.c]: please add "static" for every local functions and variables.

**Comment by seinfra** [comment]
Code Rule Check OK.

**Comment by samegautam** [comment]
@sunghan-chang  : Please review. 
In fix , we have changed testcase name as per updated coding conventions.

**Comment by sunghan-chang** [comment]
@samegautam No, please don't mix different kind of change. 
Adding "static" and changing function name for existed should be splitted from this commit.

**Comment by samegautam** [comment]
Changes would be commit in split commit.

**Pull request closed without merging**